docs/conf/scalar-prometheus-custom-values.yaml

```diff
@@ -17,6 +17,11 @@ grafana:
   defaultDashboardsEnabled: false
 
   sidecar:
+    datasources:
+      enabled: true
+      defaultDatasourceEnabled: false
```

Comment:
When I set the following two data sources, I faced some errors after restarting Grafana pods. And, the data sources are not configured properly since the error occurs.

* Data sources
  * Prometheus (using default datasource `grafana.sidecar.datasources.defaultDatasourceEnabled = true`)
  * Loki (using `grafana.additionalDataSources`)
* Configuration
  ```yaml
  grafana:
    sidecar:
      datasources:
        enabled: true
        defaultDatasourceEnabled: true
    additionalDataSources:
    - name: Loki
      type: loki
      uid: loki
      url: http://scalar-logging-loki.monitoring.svc.cluster.local:3100/
      access: proxy
      editable: false
      isDefault: false
  ```
* Error of Grafana container
  ```
  logger=provisioning t=2023-01-24T06:30:10.829402508Z level=error msg="Failed to provision data sources" error="Datasource provisioning error: datasource.yaml config is invalid. Only one datasource per organization can be marked as default"
  
  logger=context userId=1 orgId=1 uname=admin t=2023-01-24T06:30:10.829461079Z level=error msg= error="Datasource provisioning error: datasource.yaml config is invalid. Only one datasource per organization can be marked as default" remote_addr=127.0.0.1 traceID=
  
  logger=context userId=1 orgId=1 uname=admin t=2023-01-24T06:30:10.829507843Z level=error msg="Request Completed" method=POST path=/api/admin/provisioning/datasources/reload status=500 remote_addr=127.0.0.1 time_ms=10 duration=10.223829ms size=48 referer= handler=/api/admin/provisioning/datasources/reload
  ```
* Error of Grafana sidecar container (grafana-sc-datasources)
  ```
  requests.exceptions.RetryError: HTTPConnectionPool(host='localhost', port=3000): Max retries exceeded with url: /api/admin/provisioning/datasources/reload (Caused by ResponseError('too many 500 error responses'))
  ```

It seems that the `isDefault` configuration causes some problem. And, the default data source created by `grafana.sidecar.datasources.defaultDatasourceEnabled = true` is defined [here](https://github.com/prometheus-community/helm-charts/blob/6808e4cbfbec32b09323986a171b25b6999cc8e0/charts/kube-prometheus-stack/templates/grafana/configmaps-datasources.yaml#L25-L36). It sets `isDefault = true`.

So, I configured two data sources using `grafana.additionalDataSources` and disabled `grafana.sidecar.datasources.defaultDatasourceEnabled` as follows.

```yaml
grafana:
  sidecar:
    datasources:
      enabled: true
      defaultDatasourceEnabled: false
  additionalDataSources:
  - name: Prometheus
    type: prometheus
    uid: prometheus
    url: http://scalar-monitoring-kube-pro-prometheus.monitoring.svc.cluster.local:9090/
    access: proxy
    editable: false
    isDefault: false
    jsonData:
      timeInterval: 30s
  - name: Loki
    type: loki
    uid: loki
    url: http://scalar-logging-loki.monitoring.svc.cluster.local:3100/
    access: proxy
    editable: false
    isDefault: false
```

As a result, there is no error and both data sources are configured properly after restarting the Grafana pods. So, I adopt this configuration as a sample.